src/onnx/parse_slice.cpp

```diff
@@ -77,7 +86,39 @@ struct parse_slice : op_parser<parse_slice>
             op.axes = axes;
         }
 
-        return info.add_instruction(op, args[0]);
+        if(reverse_direction)
+        {
+
+            migraphx::argument axes_arg = args.at(3)->eval();
+            std::vector<int> axes_v;
+            axes_arg.visit([&](auto s) { axes_v.assign(s.begin(), s.end()); });
+
+            auto lens = args[0]->get_shape().lens();
+
+            for(auto axis : axes_v)
+            {
+                auto start_v = op.starts[axis];
+                auto end_v   = op.ends[axis];
+                if((start_v < 0) && (end_v < INT_MIN))
+                {
+                    op.ends[axis]   = lens[axis] + start_v + 1;
+                    op.starts[axis] = 0;
+                }
+                else if((start_v < 0) && (end_v > INT_MIN) && (end_v < 0))
+                {
+                    op.ends[axis]   = lens[axis] + start_v + 1;
+                    op.starts[axis] = end_v - INT_MIN;
+                }
+            }
+
+            auto ins = info.add_instruction(op, args[0]);
```

Comment:
This should be moved outside of the if statement and then only construct reverse when raxes is not empty:

```cpp
auto ins = info.add_instruction(op, args[0]);
if (not raxes.empty())
    return info.add_instruction(make_op("reverse", {{"axes", raxes}}), ins);
else
    return ins;
```

Reply:
You can easily add the step operator here:

```cpp
auto ins = info.add_instruction(op, args[0]);
if (std::any_of(steps.begin(), steps.end(), [](auto s) { return std::abs(s) != 1 }))
{
    std::vector<int64_t> nsteps;
    std::transform(steps.begin(), steps.end(), std::back_inserter(nsteps), [](auto s) {return std::abs(s); });
    ins = info.add_instruction(make_op("step", {{"axes", op.axes}, {"steps", nsteps}}));
}
if (not raxes.empty())
    return info.add_instruction(make_op("reverse", {{"axes", raxes}}), ins);
else
    return ins;
```

But this would probably entail more unit testing, so I guess we can add it later on since you need it to run a model already.

Reply:
Done

Reply:
I have created the issue #855 to do the addition of step operator later.